.ci/pitest.groovy

```diff
@@ -0,0 +1,459 @@
+    final Process proc = mavenCommand.execute()
+    proc.in.eachLine { println it }
+}
+
+/**
+ * Check the generated pitest report. Parse the survived and suppressed mutations and compare
+ * them.
+ *
+ * @param profile the pitest profile to execute
+ * @param flag command line argument flag to determine output format
+ * @return {@code 0} if pitest report is as expected, {@code 1} otherwise
+ */
+private static int checkPitestReport(String profile, String flag) {
+    final XmlParser xmlParser = new XmlParser()
+    File mutationReportFile = null
+    final String suppressedMutationFileUri =
+            ".${SEPARATOR}.ci${SEPARATOR}pitest-suppressions${SEPARATOR}${profile}-suppressions.xml"
+    new File(".${SEPARATOR}target${SEPARATOR}pit-reports")
+            .eachFileRecurse(FileType.FILES) {
+                if (it.name == 'mutations.xml') {
+                    mutationReportFile = it
+                }
+            }
+    final Node mutationReportNode = xmlParser.parse(mutationReportFile)
+    final Set<Mutation> survivedMutations = getSurvivedMutationsList(mutationReportNode)
+
+    final File suppressionFile = new File(suppressedMutationFileUri)
+    Set<Mutation> suppressedMutations = Collections.<Mutation> emptySet()
+    if (suppressionFile.exists()) {
+        final Node suppressedMutationNode = xmlParser.parse(suppressedMutationFileUri)
+        suppressedMutations = getSuppressedMutationList(suppressedMutationNode)
+    }
+    return compareMutations(survivedMutations, suppressedMutations, flag)
+}
+
+/**
+ * Get the survived mutation list. All child nodes of the main {@code mutations} node
+ * are parsed.
+ *
+ * @param mainNode the main {@code mutations} node
+ * @return A set of survived mutations
+ */
+private static Set<Mutation> getSurvivedMutationsList(Node mainNode) {
+
+    final def children = mainNode.children()
+    final Set<Mutation> survivedMutations = new TreeSet<>()
+
+    children.each { node ->
+
+        final Node mutationNode = node as Node
+
+        final String mutationStatus = mutationNode.attribute("status")
+
+        if (mutationStatus == "SURVIVED") {
+            survivedMutations.add(getMutation(mutationNode))
+        }
+    }
+    return survivedMutations
+}
+
+/**
+ * Get the suppressed mutation list. All child nodes of the main {@code suppressedMutations} node
+ * are parsed.
+ *
+ * @param mainNode the main {@code suppressedMutations} node
+ * @return A set of suppressed mutations
+ */
+private static Set<Mutation> getSuppressedMutationList(Node mainNode) {
+    final def children = mainNode.children()
+    final Set<Mutation> suppressedMutations = new TreeSet<>()
+
+    children.each { node ->
+        final Node mutationNode = node as Node
+        suppressedMutations.add(getMutation(mutationNode))
+    }
+    return suppressedMutations
+}
+
+/**
+ * Construct the {@link Mutation} object from the {@code mutation} XML node.
```

Comment:
Extend this comment to mention that we get line of source code from file itself.

Reply:
Done.